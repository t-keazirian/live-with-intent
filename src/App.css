* {
	box-sizing: border-box;
}

html,
body {
	font-family: avenir;
	height: 100%;
	background-color: #f1f2eb;
}

div.app {
	display: flex;
	flex-direction: column;
}

main.main-app {
	flex: 1 0 auto;
}

div.error {
	text-align: center;
	padding-top: 5px;
	color: red;
}

div.one-goal-item {
	margin-top: 60px;
}

div.category-div,
div.note-div {
	display: flex;
	flex-direction: column;
	align-content: center;
	text-align: center;
	width: 50%;
}

section.add-goal-section,
section.edit-goal-section {
	/* display: flex; */
	background: rgb(164, 194, 165);
	border-radius: 10px;
	color: #f2ebf1;
	border-radius: 10px;
	box-shadow: 5px 5px 5px #566246;
	margin: 10px;
}

h2.intro {
	color: rgb(164, 194, 165);
}

h4 {
	margin: 5px;
	font-weight: 900;
}

h2.goals-header {
	font-weight: 900;
}

div.goal-item,
div.goal-item-info {
	align-items: center;
	background: rgb(164, 194, 165);
	display: flex;
	flex-direction: column;
	margin: 10px 20px;
	border-radius: 10px;
	box-shadow: 5px 5px 5px #566246;
}

div.goal-item-info {
	color: #f2ebf1;
}

input[type='text'],
textarea {
	padding: 5px;
	border-radius: 5px;
	border: none;
	font-family: inherit;
}

select {
	border-radius: 5px;
	padding: 5px;
	background: white;
	font-family: inherit;
}

div.button-container {
	display: flex;
	align-content: space-around;
	margin-bottom: 10px;
}

button {
	margin: 5px;
	background-color: #f2b880;
	color: #4a4a48;
	cursor: pointer;
	border-radius: 28px;
	padding: 10px 20px;
	font-weight: bolder;
	border: none;
	box-shadow: 3px 2px 5px #ea8c34;
}

button:disabled {
	background-color: gray;
	box-shadow: none;
}

a:link {
	color: #f1f2eb;
	text-decoration: none;
}

a:visited {
	color: #f1f2eb;
}

a:hover {
	color: #4a4a48;
}

a.nav-link {
	color: #f1f2eb;
}

@media all and (min-width: 600px) {
	ul {
		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: center;
	}

	div.goal-item {
		flex-direction: row;
		align-self: center;
		padding: 10px;
	}

	div.button-container {
		margin-right: 10px;
		justify-items: center;
		align-self: center;
		margin: 0;
	}

	div.goal-item-info {
		width: 50%;
	}

	section.add-goal-section,
	section.edit-goal-section {
		width: 50%;
		align-self: center;
	}
}
