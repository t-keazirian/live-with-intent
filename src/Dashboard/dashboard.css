section.buttons {
  padding-top: 40px;
  text-align: center;
}

select#goal-type, select#goal-category, input.name-search {
  margin: 5px;
}

button.add-recurring {
  margin-top: 50px;
}

button.add-one-time {
  margin-top: 15px;
}

div.goals-filter {
  display: flex;
  justify-content: center;
  padding: 5px;
  margin-left: 15px;
  margin-right: 15px;
}

[type=submit], button {
  margin: 5px;
}

@media all and (min-width: 800px) {
  section.buttons {
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    justify-content: center;
  }

  button.add-one-time, button.add-recurring {
    margin-left: 10px;
  }

  div.goals-filter {
    flex-direction: row;
  }
}