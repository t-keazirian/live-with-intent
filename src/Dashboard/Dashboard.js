import React from 'react';
import './dashboard.css';
import TotalGoals from '../TotalGoals/TotalGoals';
import { Link } from 'react-router-dom';
import CategoryList from '../CategoryList/CategoryList';

class Dashboard extends React.Component {

	render() {
		return (
			<div className="dashboard">
				<section className="buttons">
					<Link to="/add-new-goal">
						<button className="add-one-time">Add Goal</button>
					</Link>
				</section>
				<section className="goal-summary-container">
					<div className="goal-summary">
						<h3 className="goal-header">Viewing __ Total Goals</h3>
					</div>
					<div className="goals-filter">
						<input
							type="text"
							className="name-search"
							placeholder="Search goals by name"
						/>

						<CategoryList />

					</div>
				</section>

				<TotalGoals />
			</div>
		);
	}
}

export default Dashboard;
