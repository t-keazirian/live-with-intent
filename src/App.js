import React from 'react';
import './App.css';
import Nav from './Nav/Nav';
import Footer from './Footer/Footer'
import AddNewRGoal from './AddNewRGoal/AddNewRGoal';
import { Route, Switch } from "react-router-dom";
import AddNewOTGoal from './AddNewOTGoal/AddNewOTGoal';
import Dashboard from './Dashboard/Dashboard';
import LandingPage from './LandingPage/LandingPage';
import NotFound from './NotFound/NotFound';

function App() {
  return (
    <div className='main-app'>
    <main className='App'>
    <nav>
      <Nav />
    </nav>
    <Switch>
      <Route 
        exact path='/'
        component={LandingPage}
      />
      <Route 
        path='/dashboard'
        component={Dashboard}
      />
      <Route 
        path='/add-new-recurring'
        component={AddNewRGoal}
      />
      <Route 
        path='/add-new-onetime'
        component={AddNewOTGoal}
      />
      <Route 
        component={NotFound}
      />
      </Switch>
    </main>
    <footer>
      <Footer />
    </footer>
    </div>
  );
}

export default App;