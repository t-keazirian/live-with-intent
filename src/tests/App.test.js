import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import Dashboard from '../Dashboard/Dashboard';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('App Tests', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<App />
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});
});

describe('Dashboard', () => {
	it('Dashboard renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<Dashboard />
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});
});

describe('Dashboard snapshot', () => {
	it('renders the UI as expected', () => {
		const tree = renderer.create(
    <BrowserRouter>
    <Dashboard />
    </BrowserRouter>).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
