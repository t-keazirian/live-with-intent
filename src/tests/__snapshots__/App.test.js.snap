// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapshot renders the UI as expected 1`] = `
<div
  className="dashboard"
>
  <section
    className="buttons"
  >
    <a
      href="/add-new-recurring"
      onClick={[Function]}
    >
      <button
        className="add-recurring"
      >
        Add Recurring Goal
      </button>
    </a>
    <a
      href="/add-new-onetime"
      onClick={[Function]}
    >
      <button
        className="add-one-time"
      >
        Add One-Time Goal
      </button>
    </a>
  </section>
  <section
    className="goal-summary-container"
  >
    <div
      className="goal-summary"
    >
      <h3
        className="goal-header"
      >
        Viewing __ Total Goals
      </h3>
    </div>
    <div
      className="goals-filter"
    >
      <input
        className="name-search"
        placeholder="Search goals by name"
        type="text"
      />
      <div
        className="goal-category"
      >
        <select
          id="goal-category"
          name="goal-category"
          required={true}
        >
          <option
            selected={true}
            value=""
          >
            Choose Category
          </option>
          <option
            value="get-fit"
          >
            Get Fit
          </option>
          <option
            value="nutrition"
          >
            Nutrition
          </option>
          <option
            value="personal-growth"
          >
            Personal Growth
          </option>
          <option
            value="learn-skill"
          >
            Learn A Skill
          </option>
          <option
            value="be-productive"
          >
            Be Productive
          </option>
          <option
            value="mindfulness"
          >
            Mindfulness
          </option>
          <option
            value="sleep"
          >
            Sleep
          </option>
        </select>
      </div>
      <select
        id="goal-type"
        name="goal-type"
      >
        <option
          defaultValue=""
          value=""
        >
          Search by Type
        </option>
        <option
          value="one-time"
        >
          One-Time Goals
        </option>
        <option
          value="recurring"
        >
          Recurring Goals
        </option>
      </select>
    </div>
  </section>
  <div
    className="total-goals"
  >
    <div
      className="goals-container"
    >
      <div
        className="goals-list"
      >
        <div
          className="goal-item"
        >
          <h2
            className="goals-header"
          >
            Brush Teeth
          </h2>
          <div>
            <span
              className="category"
            >
              Get Fit
            </span>
          </div>
          <div
            className="goal-note"
          >
            <span>
              Brush teeth 2x per day
            </span>
          </div>
          <div
            className="button-container"
          >
            <button
              type="submit"
            >
              Edit
            </button>
            <button
              type="submit"
            >
              Delete
            </button>
          </div>
        </div>
        <div
          className="goal-item"
        >
          <h2
            className="goals-header"
          >
            Journal
          </h2>
          <div>
            <span
              className="category"
            >
              Personal Growth
            </span>
          </div>
          <div
            className="goal-note"
          >
            <span>
              Journal 1x per day
            </span>
          </div>
          <div
            className="button-container"
          >
            <button
              className="edit-goal-btn"
              type="submit"
            >
              Edit
            </button>
            <button
              className="delete-goal-btn"
              type="submit"
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
